react/features/app-settings/components/AbstractAppSettings.js

```diff
@@ -108,15 +116,55 @@ export class AbstractAppSettings extends Component<Props> {
         });
     }
 
+    _onFinalizeServerURL: () => boolean;
+
+    /**
+     * Cleans up server URL when the user leaves the field or the screen.
+     * This is required since we want to let the user to freely edit the field
+     * without any limitation at edit time.
+     *
+     * @protected
+     * @returns {boolean}
+     */
+    _onFinalizeServerURL() {
+        let { serverURL } = this.props._profile;
+
+        if (serverURL) {
+            const urlRegExp = new RegExp('^(\\w+(?:s)?://)?(.+)$');
+            const urlComponents = urlRegExp.exec(serverURL.toLowerCase());
```

Comment:
can you please normalize the URL by removing spaces and lower casing it **before** the checks? otherwise this fails: "htt   p:// lalala.com".